projects/ADuCM360_demo_cn0409/src/main.c

```diff
@@ -0,0 +1,203 @@
+/************************* Functions Definitions ******************************/
+/******************************************************************************/
+
+/**
+  @brief UART Interrupt Handler
+
+  @return none
+
+**/
+void UART_Int_Handler (void)
+{
+	unsigned short  status;
+	char c;
+
+	status = UrtIntSta(pADI_UART);                      /* Check UART status */
+
+	if (status & COMIIR_NINT) {
+		return;   /* Check if UART is busy */
+	}
+
+	switch (status & COMIIR_STA_MSK) {                  /* Check what command to execute */
+	case COMIIR_STA_RXBUFFULL:                     /* Check if UART register is available to be read */
+
+		UART_ReadChar(&c);                          /* Read character from UART */
+
+		if( c == 'y') {                              /* Check if read character is ENTER */
+			uart_cmd = UART_TRUE;                    /* Set flag */
+			uart_received = UART_TRUE;
+		} else {
+			if ( c == 'n') {
+				uart_cmd = UART_FALSE;
+				uart_received = UART_TRUE;
+			}
+		}
+		if (c == 0x0D) {
+			//enter received
+			uart_received = UART_TRUE;
+			uart_enter = UART_TRUE;
+			s[_index] = '\0';
+			_index = 0;
+		}
+		if (((c >= '0') && (c <= '9')) | (c == '.')) {
+			s[_index] = c;
+			_index++;
+		}
+		break;
+
+
+	case COMIIR_STA_TXBUFEMPTY:                                    /* Check if UART register is available to be written */
+
+		if (uart_tcnt) {                                             /* Check uart counter */
+
+			uart_tbusy = UART_TRUE;                                    /* UART is busy with writing*/
+
+			uart_tcnt--;                                               /* Decrease  uart counter */
+
+			UART_WriteChar(uart_tx_buffer[uart_tpos++], UART_WRITE);   /* Write character to UART */
+
+			if (uart_tpos == UART_TX_BUFFER_SIZE) {                    /* Check if TX buffer is full */
+				uart_tpos = 0;                                           /* Reset buffer counter  */
+			}
+
+		} else {
+
+			uart_tbusy = UART_FALSE;                               /* UART is no longer busy with writing */
+		}
+
+		break;
+
+	default:
+		;
+	}
+
+}
+
+/**
+  @brief I2C0 Interrupt Handler
+
+  @return none
+
+**/
+void I2C0_Master_Int_Handler(void)
+{
+
+	ui16I2cStatus = 0;
+	ui16I2cStatus = I2cSta(MASTER);
+
+	if((ui16I2cStatus & I2CMSTA_RXREQ) == I2CMSTA_RXREQ) { // Master Receive IRQ
+		uiMasterRxDat[uiMasterRxIndex++] = I2cRx(MASTER);
+	}
+}
+
+
+int main(int argc, char* argv[])
+{
+	uint16_t au16Registers[128];
+
+	timer_start ();
+
+	/* Initialize UART port */
+	UART_Init (B9600, COMLCR_WLS_8BITS);
+	/* Initialize I2C communication */
+	I2C_Init();
+
+	AppPrintf("Test for turbidity\n\r");
+	AppPrintf("Device id for ADPD105: %04x\n\r", ADPD105_ReadReg(ADPD105_DEVID));
+
+	/*Init and configure CN0358*/
+	CN0409_Init();
+	CN0409_AFECalibration();
+	CN0409_ChannelOffsetCalibration();
+
+	// Infinite loopy
+
+	while (1) {
+		CN0409_InteractiveMenu();
+
+		timer_sleep(3000);
+	}
+	// Infinite loop, never return.
```

Comment:
decide for one type of comment either `//` or `/*`

Reply:
Please use C style comments, I think that is more consistent with other projects we do, isn't it?